### PR TITLE
[Nextjs] Preserve default locale in external absolute urls

### PR DESCRIPTION
**Pull request by MenKNas**
- [ ] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
Fixes an issue where the default locale was stripped from external absolute URLs in redirects.

Example:
Before: `/en-hk/ra` → `https://example.com/this-is-en`
After: `/en-hk/ra` → `https://example.com/en/this-is-en`

## Testing Details
- [x] Unit Test Added
- [ ] Manual Test/Other (Please elaborate)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Our versioning strategy is as follows:
 
 ### 🐛 Bug Fixes
 
+* `[Next.js]` Preserve default locale in external absolute urls ([#2142](https://github.com/Sitecore/jss/pull/2142))
```
Comment (illiakovalenko) [inline CHANGELOG.md]: Can we mention, please, the area that was affected and fixed? It's related to redirects, but the title doesn't mention that

File: packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts [hunk 2]
```diff
@@ -104,9 +101,9 @@ export class RedirectsMiddleware extends MiddlewareBase {
             let patternPath = urlArray[0].toLowerCase();
             // nextjs routes are case-sensitive, but locales should be compared case-insensitively
             const patternParts = patternPath.split('/');
-            const maybeLocale = patternParts[1].toLowerCase();
+            const maybeLocale = patternParts[1]?.toLowerCase?.();
             // case insensitive lookup of locales
-            if (new RegExp(this.locales.join('|'), 'i').test(maybeLocale)) {
+            if (maybeLocale && new RegExp(this.locales.join('|'), 'i').test(maybeLocale)) {
```
Comment (illiakovalenko) [inline packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts]: Overall looks like old implementation worked fine as well, do we need to apply these changes? is there is an edge case when `patternParts` is empty, so it throws an error?

File: packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts [hunk 3]
```diff
@@ -229,13 +224,22 @@ export class RedirectsMiddleware extends MiddlewareBase {
...
+          true
+        );
       } else {
         const isUrl = isRegexOrUrl(existsRedirect.pattern) === 'url';
         const targetParts = existsRedirect.target.split('/');
         const urlFirstPart = targetParts[1];
 
-        if (this.locales.includes(urlFirstPart)) {
+        if (
+          !REGEXP_ABSOLUTE_URL.test(existsRedirect.target) &&
```
Comment (illiakovalenko) [inline packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts]: This change should be reverted

File: packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.test.ts
Comment (illiakovalenko) [inline packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.test.ts:1]: We need to add a unit test to cover the edge case when absolute url is used, like it was done [here](https://github.com/Sitecore/content-sdk/pull/201/files#diff-bb611780685e8aa493f117051b376d92bccf5fba86a4f97f2f45341bb569fe7cR1726)

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts (hunk 2 above)
Comment (MenKNas) [reply packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts]: I will revert it since it worked fine—no reason to extend the scope of this too much.

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,6 +17,7 @@ Our versioning strategy is as follows:
 
 ### 🐛 Bug Fixes
 
+* `[Next.js]``[Redirects]` Preserve default locale in external absolute urls ([#2142](https://github.com/Sitecore/jss/pull/2142))
```
Comment (illiakovalenko) [inline CHANGELOG.md]: Let's add a whitespace, otherwise "preview" is broken
```suggestion
* `[Next.js]` `[Redirects]` Preserve default locale in external absolute urls ([#2142](https://github.com/Sitecore/jss/pull/2142))
```

**Review by illiakovalenko — APPROVED** [review]
Looks good! 👍
